### PR TITLE
[Refactor] ClientConn always returns a working connection

### PR DESCRIPTION
**Pull request by neha-viswanathan**

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -96,15 +96,15 @@ func (c *cli) OnInitialize(initializers ...func()) {
...
 
 // ClientConn returns the grpc connection to the API.
-func (c *cli) ClientConn() (*grpc.ClientConn, error) {
+func (c *cli) ClientConn() *grpc.ClientConn {
 	if c.clientConn == nil {
 		var err error
 		c.clientConn, err = NewClientConn(c.Server(), GetToken())
 		if err != nil {
-			return nil, err
+			c.Console().Fatalf("error while establishing grpc connection : %s", err)
```
Comment (bquenin) [inline cli/cli.go]: Should be `c.Console().Fatalf("error while establishing grpc connection: %s\n", err)`

File: cli/command/version/version.go [hunk 2]
```diff
@@ -75,20 +75,18 @@ func showVersion(c cli.Interface) error {
...
-			return fmt.Errorf("%s", grpc.ErrorDesc(err))
-		}
-		v.Server = &version.Info{
-			Version:   reply.Info.Version,
-			Build:     reply.Info.Build,
-			GoVersion: reply.Info.GoVersion,
-			Os:        reply.Info.Os,
-			Arch:      reply.Info.Arch,
-		}
+	conn := c.ClientConn()
```
Comment (bquenin) [inline cli/command/version/version.go]: amp version doesn't work as before (when the server is not reachable):
```
$ amp --server localhost:50101 version
grpc: addrConn.resetTransport failed to create client transport: [connection error: desc = "transport: dial tcp 127.0.0.1:50101: getsockopt: connection refused" {localhost:50101 <nil>}]; Reconnecting to %!v(MISSING)Fatal: grpc: addrConn.resetTransport failed to create client transport: [connection error: desc = "transport: dial tcp: operation was canceled" {localhost:50101 <nil>}]; Reconnecting to %!v(MISSING)error while establishing grpc connection : context deadline exceeded%                                                                                                                                                                                                                                        
```

It should be:
```
$ amp --server localhost:50101 version
grpc: addrConn.resetTransport failed to create client transport: [connection error: desc = "transport: dial tcp 127.0.0.1:50101: getsockopt: connection refused" {localhost:50101 <nil>}]; Reconnecting to %!v(MISSING)Client:
 Version:       v0.6.0-dev
 Build:         a64db8e7
 Server:        localhost:50101
 Go version:    go1.8.1
 OS/Arch:       linux/amd64

Server:         not connected
grpc: addrConn.resetTransport failed to create client transport: [connection error: desc = "transport: dial tcp: operation was canceled" {localhost:50101 <nil>}]; Reconnecting to %!v(MISSING)Error: context deadline exceeded
```

**Review by bquenin — CHANGES_REQUESTED** [review]
Good job, some fixes needed

File: cli/cli.go (hunk 1 above)
Comment (subfuzion) [inline cli/cli.go]: `Fatalf` will prepend 'Fatal' to the message output, so maybe `unable to establish grpc connection: %s\n` is a better message. And don't forget to add a newline ... or just use `c.Console().Fatalln("unable... : ", err.Error())` instead.

**Review by subfuzion — CHANGES_REQUESTED** [review]
Thanks for this @neha-viswanathan ... what I wanted. ;) Looks good to me with one request (and also make sure passes @bquenin's sense of aesthetics as well)! :p

File: cli/command/version/version.go (hunk 2 above)
Comment (subfuzion) [reply cli/command/version/version.go]: I pushed an update to get this expedited. Now we have `Connect()` and `ClientConn` methods. Who says you can't have your cake and eat it, too?

Not connected (note extra output due to grpc logging being enabled):
```
$ amp --server localhost:50000 version
grpc: addrConn.resetTransport failed to create client transport: [connection error: desc = "transport: dial tcp [::1]:50000: getsockopt: connection refused" {localhost:50000 <nil>}]; Reconnecting to %!v(MISSING)
grpc: addrConn.resetTransport failed to create client transport: [connection error: desc = "transport: dial tcp: operation was canceled" {localhost:50000 <nil>}]; Reconnecting to %!v(MISSING)
Client:
 Version:       v0.6.0-dev
 Build:         48c183ac
 Server:        localhost:50000
 Go version:    go1.8
 OS/Arch:       darwin/amd64

Server:         not connected
```

Connected:
```
$ amp --server localhost:50101 version
Client:
 Version:       v0.6.0-dev
 Build:         48c183ac
 Server:        localhost:50101
 Go version:    go1.8
 OS/Arch:       darwin/amd64

Server:         
 Version:       v0.6.0-dev
 Build:         7f736a11
 Go version:    go1.8.1
 OS/Arch:       linux/amd64
```

Comment (subfuzion) [reply cli/command/version/version.go]: @bquenin @neha-viswanathan ^^

**Review by subfuzion — APPROVED** [review]

**Pull request merged**